WTG.Analyzers.Test/TestData/ArrayAnalyzer/NoArrayEmpty/Source.cs

```diff
@@ -0,0 +1,19 @@
+public static class Foo
+{
+	public static void StaticArrayCreator<T>()
+	{
+		Method(new int[0]);
+		Method(new int[] { });
+	}
+
+	public static void Method(int[] argument)
+	{
+	}
+}
+
+namespace System
+{
+	public class Array
```

Comment:
Since the tests are passing, I assume `GetTypeByMetadataName` will prefer types defined in the compiled source and fall back to types defined in referenced assemblies, but is it contractual? or subject to change in future?

Reply:
I don’t know. I assume they wouldn’t intentionally introduce a breaking change here. Want me to ask on Gitter?

Reply:
Probably should, though I suppose it doesn't really need to block the merge since it's only tests that are affected and not production code.

Reply:
The wording on https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.compilation.gettypebymetadataname?view=roslyn-dotnet suggests/implies the same search order that we're relying on, though it doesn't explicitly define it.